lib/tasks/translate.rake

```diff
@@ -0,0 +1,19 @@
+namespace :vue do
+  desc "Translate Rails' locale into Vue's locale"
+  task :translate do
+    Dir.foreach("config/locales/").each do |locale|
+      next if locale == '.' or locale == '..'
+      config = YAML.load_file("config/locales/#{locale}")
+      name = locale.to_s.split('.').first if locale.to_s.split('.') != nil
+      if name != nil
+        hash = Hash.new
+        hash[name] = config[name]
+
+        File.open("app/javascript/locales/#{name}.json", 'w') { |file| 
+          file.puts JSON.pretty_generate(hash)
+           $stdout.puts "Created app/javascript/locales/#{name}.json"
+        }
+      end
+    end
+  end
+end
```

Comment:
💡 https://github.com/Microsoft/vscode/issues/12076

Does this sync the full rails locales, with all nodes, to a js object? Do you think this is ok, or just a subset of this data would be necessary. (All nodes in `vuejs:` e.g)
This approach in general, violates the DRY principle of rails, you basically duplicate the contents to a second location. [I've done a lot of i18n in rails and know how complex this could become.]

How about make the locales available by a rails helper in the layout so vue can somehow access by a specific `#id` – I don't know if this is technically possible – it just would be a solution to avoid duplicate the i18n key/value.

Reply:
I considered DRY and concluded differently. 

I think DRY is already violated in a multi-architecture with the two i18n systems:  Vue’s `vue-i18n` and Rails `i18n` system. 

1. Rake task with two i18n datasources:
![img_0403](https://user-images.githubusercontent.com/323891/49443040-45d6d700-f806-11e8-98b4-ff620152496d.JPG)

2. Or a global javascript object to couple one datasource to another system:
![img_0402](https://user-images.githubusercontent.com/323891/49443137-90f0ea00-f806-11e8-9b51-e3564dedf1a8.JPG)

---

## More reasons

1a. If i include a global javascript, this will add tight coupling at global level. it also violates separation of concerns. (I treat vue-i18n and rails’ i18n as two concerns since they are used for vue component and rails view separately). 

![img_0406](https://user-images.githubusercontent.com/323891/49443249-e7f6bf00-f806-11e8-8fe2-7d175b2c05a2.JPG)



2a. We both agree to reduce complexity. 

I think to reduce complexity, maintaining separation of concerns is more important than reducing  datasource duplicates. 

Furthermore, this is datasource duplication, *not code duplication* . Does DRY applies to data duplicates? 

Also,  data duplicates are fine because corruption of data can be fixed with a one-way rake task. However, a coupling of code is more serious than corruption of data. Do you agree?

![img_0405](https://user-images.githubusercontent.com/323891/49443324-1e343e80-f807-11e8-9c00-5cb42c659def.JPG)


cc @rlafranchi

Reply:
Is there a way to automate `vue:translate`? somehow hook into the build process?  If so, I think that is fine so it saves you the step of building the json file manually.

Reply:
Thanks for challenging it @m5o

Reply:
Automated rake may be achieved through guard https://github.com/guard/guard#group in fact, i think it's a good idea to have guard in Vails :development

Please take a stab if you are up to it.